crypto/property/property.c

```diff
@@ -355,9 +404,17 @@ int ossl_method_store_add(OSSL_METHOD_STORE *store, const OSSL_PROVIDER *prov,
             && tmpimpl->properties == impl->properties)
             break;
     }
-    if (i == sk_IMPLEMENTATION_num(alg->impls)
-        && sk_IMPLEMENTATION_push(alg->impls, impl))
```

Comment:
Why this change? It looks like we are now unconditionally pushing to the stack regardless of whether there is one there or not - which was the whole point of the previous loop?